tests/shared/src/test/scala/spire/syntax/CforSuite.scala

```diff
@@ -100,17 +101,21 @@ class CforSuite extends munit.FunSuite {
   }
 
   test("capture value in closure") {
-    val b1 = collection.mutable.ArrayBuffer.empty[() => Int]
+    val capturedValues = collection.mutable.ArrayBuffer.empty[Int]
+    val b1 = collection.mutable.ArrayBuffer.empty[() => Unit]
     cfor(0)(_ < 3, _ + 1) { x =>
-      b1 += (() => x)
+      val value = x // Explicitly define. Avoids failures on certain JDKs
+      val closure = () => { capturedValues += value; () }
+      b1 += closure
```

Comment:
Are you sure this is related to JDK version? This is actually due to a change in Scala 3.
- https://github.com/typelevel/spire/issues/1225

Reply:
Interesting. It was definitely passing on older JDK's. Unless I am very confused and was running on scala 2.13!

Reply:
Anyway I believe this implementation is functionally equivalent no? Replaced the while loop with a cfor loop for b2. Can revert that.

Reply:
To clarify, this is test code, not implementation code. You can consider test code to be representing user code. So if we have to change our test code because the behavior of the implementation changed, that means our users will have to change their code as well. That's not good 😅 when users upgrade, they could be inadvertently introducing bugs.